Avoid gcc-14 compiler errors:

utils.c:195:35: error: initialization of 'struct _iconv_info **' from
 incompatible pointer type 'struct _iconv_info *'
 [-Wincompatible-pointer-types]
  195 |         static iconv_t *iconv_h = (iconv_t) - 1;
      |                                   ^
utils.c:205:30: error: assignment to 'struct _iconv_info **' from
 incompatible pointer type 'iconv_t' {aka 'struct _iconv_info *'}
 [-Wincompatible-pointer-types]
  205 |                 if ((iconv_h = iconv_open("UTF-8", g_codepage)) == (iconv_t) - 1)
      |
utils.c:217:19: error: passing argument 1 of 'iconv' from incompatible pointer type [-Wincompatible-pointer-types]
  217 |         if (iconv(iconv_h, (char **) &src, &is, &dest, &os) == (size_t) - 1)
      |                   ^~~~~~~
      |                   |
      |                   struct _iconv_info **
In file included from utils.c:24:
/usr/include/iconv.h:97:23: note: expected 'iconv_t' {aka 'struct _iconv_info
 *'} but argument is of type 'struct _iconv_info **'
   97 | extern size_t   iconv(iconv_t, char **_RESTRICT_KYWD,
      |                       ^~~~~~~
utils.c:219:29: error: passing argument 1 of 'iconv_close' from incompatible
 pointer type [-Wincompatible-pointer-types]
  219 |                 iconv_close(iconv_h);
      |                             ^~~~~~~
      |                             |
      |                             struct _iconv_info **
/usr/include/iconv.h:76:29: note: expected 'iconv_t' {aka 'struct _iconv_info
 *'} but argument is of type 'struct _iconv_info **'
   76 | extern int      iconv_close(iconv_t);
      |                             ^~~~~~~
utils.c:220:25: error: assignment to 'struct _iconv_info **' from incompatible
 pointer type 'struct _iconv_info *' [-Wincompatible-pointer-types]
  220 |                 iconv_h = (iconv_t) - 1;
      |                         ^


Need to figure out why this doesn't happen on other platforms and
possibly submit upstream

diff --git rdesktop-1.9.0/utils.c b/rdesktop-1.9.0/utils.c
index f5fe628..0880fa7 100644
--- rdesktop-1.9.0/utils.c
+++ rdesktop-1.9.0/utils.c
@@ -192,7 +192,7 @@ utils_mkdir_p(const char *path, int mask)
 int
 utils_locale_to_utf8(const char *src, size_t is, char *dest, size_t os)
 {
-	static iconv_t *iconv_h = (iconv_t) - 1;
+	static iconv_t iconv_h = (iconv_t) - 1;
 	if (strncmp(g_codepage, "UTF-8", strlen("UTF-8")) == 0)
 		goto pass_trough_as_is;
 
